File: src/test/java/io/transwarp/logparser/util/FileParserTest.java
package io.transwarp.logparser.util;

import io.transwarp.logparser.Parser;
import org.junit.Test;

import java.nio.file.Paths;
import java.util.List;

/**
 * Author: stk
 * Date: 2018/4/2
 */
public class FileParserTest {
    @Test
    public void parseFolder() {
        FileParser fileParser = new FileParser();
        List<Record> records = fileParser.parseFile(Paths.get("C:\\Users\\stk\\Downloads\\logs\\elasticsearch\\elasticsearch-2018-03-11.log"));
        Parser parser = new Parser();
        parser.generate(records, "C:\\Users\\stk\\Downloads\\logs\\test-result.log");
    }
}

File: src/test/scala/io/transwarp/logparser/util/FileParserTest.scala
package io.transwarp.logparser.util

import java.io.File

import io.transwarp.logparser.Parser
import org.junit.Test

/**
  * Author: stk
  * Date: 2018/4/4
  */
class FileParserTest {
  @Test
  def parseFile(): Unit = {
    Parser.generate(FileParser.parseFile(new File("C:\\Users\\stk\\Downloads\\test-data\\logs\\elasticsearch\\elasticsearch-2018-03-19.log")),
      "C:\\Users\\stk\\Downloads\\test-data\\logs\\test-result.log")
  }
}
